src/Cassandra.IntegrationTests/Core/ClientTimeoutTests.cs

```diff
@@ -178,5 +178,33 @@ public void Should_Throw_NoHostAvailableException_When_All_Hosts_Down()
                 testCluster.ResumeNode(2);
             }
         }
+
+        /// Tests a TimeoutException is raised when all hosts down with default read timeout
+        ///
+        /// Should_Abort_After_Default_ReadTimeoutMillis tests that the driver throws a TimeoutException and aborts the 
+        /// query execution when all nodes in the cluster is down, given a the default ReadTimeoutMillis of 12 seconds.
+        ///
+        /// @since 3.0.0
+        /// @jira_ticket CSHARP-362
+        /// @expected_result A TimeoutException should be raised after 12 seconds.
+        ///
+        /// @test_category connection:timeout
+        [Test]
+        public void Should_Abort_After_Default_ReadTimeoutMillis()
+        {
+            var testCluster = TestClusterManager.GetNonShareableTestCluster(2, 1, true, false);
+            var builder = Cluster.Builder().AddContactPoint(testCluster.InitialContactPoint);
+            using (var cluster = builder.Build())
+            {
+                var session = cluster.Connect();
+                //warmup
+                TestHelper.Invoke(() => session.Execute("SELECT key FROM system.local"), 10);
+                testCluster.PauseNode(1);
+                testCluster.PauseNode(2);
+                Assert.Throws<TimeoutException>(() => session.Execute("SELECT key FROM system.local"));
```

Comment:
`TimeoutException` is thrown at when the amount of time passed and the `Task` did not complete and can be tuned using `QueryAbortTimeout` (default: 20secs).

To test `SocketOptions.ReadTimeoutMillis`,  you should set a lower readtimeout (something that multiplied by the amount of hosts, will be lower than `QueryAbortTimeout`. This situation is already covered in 

- `Should_Throw_OperationTimedOutException_When_Retry_Is_False`: Without retrying the next host, it throws an `OperationTimedOutException`.
- `Should_Throw_NoHostAvailableException_When_All_Hosts_Down`: Retries to the next host, but also times out, it throws a `NoHostAvailableException` with `OperationTimedOutException` as (child) inner errors per host.